### ISSUE TITLE
SDK limits query size to 30009, not 35000

### ISSUE THREAD

**Issue by dtbuchholz** [tablelandnetwork/js-tableland#382]
**Describe the bug**
I'm writing to a table while using `local-tableland`. I'm hitting a max query size of 30009, which differs from the registry's imposed limit of 35000. Note that the [`wasm-sqlparser`](https://github.com/tablelandnetwork/wasm-sqlparser) is what checks query length, not the SDK.

**To Reproduce**
Create a React app with TS: `npx create-react-app --template typescript`

Excuse the naivety. I was trying to figure out what the limit imposed was. The code has 2 sections—the first is the setup, and the second is just the write query & how the 30009 length was generated.

The code below summarizes the impl; see the full code and place it in `App.tsx` here: https://gist.github.com/dtbuchholz/c57d8f63b91c5fb02ac57d3d29ae8384

```typescript
// Define schema
interface TableSchema {
  id: number;
  name: string;
}
// Connect to local-tableland via .env private key
const privateKey = process.env.REACT_APP_PRIVATE_KEY;
let signer;
if (privateKey !== undefined) {
  const wallet = new Wallet(privateKey);
  const provider = getDefaultProvider("http://127.0.0.1:8545");
  signer = wallet.connect(provider);
}
const db = new Database<TableSchema>({ signer });
// Create a table
const prefix: string = "my_sdk_table";
const { meta: create } = await db
  .prepare(`CREATE TABLE ${prefix} (id integer primary key, name text);`)
  .run();
const name = create.txn?.name;

// Write to the table -- here, 30009 is the limit. Simply add a character in the final concat to increase to 30010 and fail
let s = "";
for (let i = 0; i < 4992; i++) {
  s += "('a'),";
}
s += "('aaaaa')"; // Add a character here to increase to 30010
const statement = `INSERT INTO ${name} (name) VALUES ${s};`;
let utf8Encode = new TextEncoder().encode(statement); // Check the bytes size
console.log(utf8Encode.length);
let blobLength = new Blob([statement]);
console.log(blobLength.size); // Check the bytes size using a different way
const { meta: insert } = await db.prepare(statement).run();
await insert.txn?.wait(); // Btw, this throws a 404: GET http://localhost:8080/api/v1/receipt/31337/0xa9332f378d89ac64424b783d1b2124d9845865f969d3e05f67f05d66b9292882?tableId=3&blockNumber=3035 404 (Not Found)
```

**Expected behavior**
Allow for queries that are 35000 in length, i.e., 30010 to 35000 should not fail

**Desktop (please complete the following information):**

- OS: Mac M1
- Browser: Chrome—connection was through .env though, not MM / browser wallet
- SDK: 4.0.0-pre.6
- local-tableland: 1.0.0-pre.2
- Node: v18.12.1
- npm: 9.2.0

**Comment by joewagner** [comment]
@dtbuchholz I'm able to reproduce, and it looks like the wasm-sqlparser lib might not be checking the statement size.  It appears to instead be checking the abstract syntax tree length.  I'll work on fixing this now.

**Comment by dtbuchholz** [comment]
Interesting, nice!

**Comment by carsonfarmer** [comment]
No it isn't checking the AST size, but it _is_ checking the size of the _parsed_ SQL statement, which, depending on the SQL statement, may or may not be the expected size. The parser adds statement components and canonicalizes the string in some cases. One way to check this is to use the parser directly and see what it spits out from @dtbuchholz, query string?

**Comment by carsonfarmer** [comment]
```typescript
import { helpers } from "@tableland/sdk";
const { normalize } = helpers;

let s = "";
for (let i = 0; i < 4992; i++) {
  s += "('a'),";
}
s += "('aaaaa')";

const statement = `INSERT INTO some_table_name (name) VALUES ${s};`;

const parsed = await normalize(statement)
const total = parsed.join(";")

let encoded = new TextEncoder().encode(total);
console.log(encoded.byteLength);
```

**Comment by carsonfarmer** [comment]
If I run the above test, I see that the original statement is 30004 bytes, whereas the parsed statement is 34995. This is still slightly below the expected cutoff of 35000, which makes sense. If you add some more "loop" to the above, it should end up too long.

**Comment by carsonfarmer** [comment]
One "solution" or fix here might just be a better error message that includes the fact that it is the "parsed" statement output that is too long?

**Comment by joewagner** [comment]
Yep, looks like I spoke to soon. @carsonfarmer I can confirm the wasm-parser is checking the statement length. The sql parser is adding length to the statement.  Doing a few tests (I could be wrong) makes it look like "github.com/tablelandnetwork/sqlparser" adds spaces to statements that include comma separated lists of values.

Parsing:
`INSERT INTO my_sdk_table_31337_3 (name) VALUES ('a'),('a'),('a');`
results in:
`insert into my_sdk_table_31337_3 (name) values ('a'), ('a'), ('a');`

**Comment by carsonfarmer** [comment]
Yes that appears to be the case, it also adds a space between multi-statement calls (not the case in this specific issue though).

**Comment by dtbuchholz** [comment]
So, would it be possible for the parser to _not_ include a space in multi-statement calls? And in this example, parse `insert into my_sdk_table_31337_3 (name) values ('a'), ('a'), ('a');` as `insert into my_sdk_table_31337_3(name) values('a'),('a'),('a');`?

Idk how the parser needs to work under the hood, but every character counts in terms of cost reduction once that query is sent to the registry.

**Comment by dtbuchholz** [comment]
> Something like this
> ```
> ...
> const { statements } = await normalize(statement);
> const total = statements.join(";");
>
> encoded = new TextEncoder().encode(total);
> console.log(encoded.byteLength);

Fwiw, that code snippet also confirmed the query size to be 35000.

**Comment by joewagner** [comment]
I believe this is fixed via 3d3997b
@dtbuchholz are we ok closing this?
